### PR TITLE
Update require build tools in README

### PR DESCRIPTION
**Pull request by omus**
I've been building Julia 0.5 from source in a Docker container and noticed that the required build tools needed a slight update:
- LLVM fails to build unless you have Python 2.7+ installed:
- cmake fails to build libgit2 unless the version is at least 3.1.2.
  For details see: https://cmake.org/Bug/view.php?id=15386

### CODE REVIEW

**Comment by tkelman** [comment]
the issue with cmake is linux specific, so the minimum version isn't the same everywhere

**Comment by omus** [comment]
@tkelman fair criticism. Would changing it to `cmake (>= 3.1.2 on Linux)` be acceptable?

**Comment by vtjnash** [comment]
That sounds good to me. As a future note, when we switch to llvm-3.9+, we will need to update this again to say CMake >= 3.4.3

**Comment by tkelman** [comment]
It's not true for all Linux distros though, only those with certain versions of openssl. Maybe just put this as a build troubleshooting note along with what the error looks like.

**Comment by omus** [comment]
@tkelman I've added a row to build troubleshooting. I've left in the `>= 3.1.2` for now as this version of CMake is rather old and it will only help users avoid this issue.

File: README.md [hunk 1]
```diff
@@ -217,6 +217,7 @@ Install or contact your systems administrator to install a more recent version o
...
 Illegal Instruction error | Check if your CPU supports AVX while your OS does not (e.g. through virtualization, as described in [this issue](https://github.com/JuliaLang/julia/issues/3263)).
+<a name="LibGit2-Build-Failure"/> LibGit2 build failure | If LibGit2 configuration step reports `CMake Error at /usr/share/cmake-3.0/Modules/FindOpenSSL.cmake:294 (list):` then the version [OpenSSL and CMake are probably incompatible](https://cmake.org/Bug/view.php?id=15386). The recommended solution is to upgrade the version of CMake to 3.1.2 or above.
```
Comment (omus) [inline README.md]: Nice catch

**Comment by omus** [comment]
Fair enough. I'll remove the version requirement out of the build tool section

**Pull request merged**